lib/ansible/modules/cloud/amazon/aws_secret.py

```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/python
+
+# Copyright: (c) 2018, Aaron Smith <ajsmith10381@gmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = r'''
+---
+module: aws_secret
+short_description: Manage secrets stored in AWS Secrets Manager.
+description:
+    - Create, update, and delete secrets stored in AWS Secrets Manager.
+author: "Aaron Smith (@slapula)"
+version_added: "2.7"
+requirements: [ 'botocore>=1.10.0', 'boto3' ]
+options:
+  name:
+    description:
+    - Friendly name for the secret you are creating.
+    required: true
+  state:
+    description:
+    - Whether the secret should be exist or not.
+    default: 'present'
+    choices: ['present', 'absent']
+  client_request_token:
+    description:
+    - UUID used to ensure idempotency.
+    - This field is autopopulated if not provided.
+  description:
+    description:
+    - Specifies a user-provided description of the secret.
+  kms_key_id:
+    description:
+    - Specifies the ARN or alias of the AWS KMS customer master key (CMK) to be
+      used to encrypt the `secret_string` or `secret_binary` values in the versions stored in this secret.
+  secret_type:
+    description:
+    - Specifies the type of data that you want to encrypt.
+    required: true
+    choices: ['binary', 'string']
+  secret:
+    description:
+    - Specifies string or binary data that you want to encrypt and store in the new version of the secret.
+    required: true
+  tags:
+    description:
+    - Specifies a list of user-defined tags that are attached to the secret.
+  rotation_lambda:
+    description:
+    - Specifies the ARN of the Lambda function that can rotate the secret.
+  rotation_interval:
+    description:
+    - Specifies the number of days between automatic scheduled rotations of the secret.
+    default: 30
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+
+EXAMPLES = r'''
+- name: Add string to AWS Secrets Manager
+  aws_secret:
+    name: 'test_secret_string'
+    state: present
+    secret_type: 'string'
+    secret: "{{ super_secret_string }}"
+
+- name: remove string from AWS Secrets Manager
+  aws_secret:
+    name: 'test_secret_string'
+    state: absent
+    secret_type: 'string'
+    secret: "{{ super_secret_string }}"
+'''
+
+
+RETURN = r'''
+secret_arn:
```

Comment:
You could probably return the contents of describe_secret() after using camel_dict_to_snake_dict (from ansible/module_utils/common/dict_transformations.py) and after using boto3_tag_list_to_ansible_dict on the tags if they exist, rather than limiting it to the secret_arn and version.